cmd/mirror.go

```diff
@@ -470,50 +470,60 @@ func yankComp(repo, id, version string) error {
 
 // the `mirror clone` sub command
 func newMirrorCloneCmd() *cobra.Command {
-	options := repository.CloneOptions{
-		Components: map[string]*[]string{},
-	}
 	var (
+		options    = repository.CloneOptions{Components: map[string]*[]string{}}
 		components []string
 		repo       *repository.V1Repository
 	)
 
-	env, fetchErr := environment.InitEnv(repoOpts)
-	if fetchErr == nil {
-		repo = env.V1Repository()
-		var index *v1manifest.Index
-		index, fetchErr = repo.FetchIndexManifest()
-
-		if index != nil && len(index.Components) > 0 {
-			for name := range index.Components {
-				components = append(components, name)
-			}
-		}
-		sort.Strings(components)
-	}
-
 	cmd := &cobra.Command{
 		Use: "clone <target-dir> [global version]",
 		Example: `  tiup mirror clone /path/to/local --arch amd64,arm --os linux,darwin    # Specify the architectures and OSs
   tiup mirror clone /path/to/local --full                                # Build a full local mirror
   tiup mirror clone /path/to/local --tikv v4  --prefix                   # Specify the version via prefix
   tiup mirror clone /path/to/local --tidb all --pd all                   # Download all version for specific component`,
-		Short:        "Clone a local mirror from remote mirror and download all selected components",
-		SilenceUsage: true,
+		Short:              "Clone a local mirror from remote mirror and download all selected components",
+		SilenceUsage:       true,
+		DisableFlagParsing: true,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			env := environment.GlobalEnv()
+			repo = env.V1Repository()
+			index, err := repo.FetchIndexManifest()
+			if err != nil {
+				return err
+			}
+
+			if index != nil && len(index.Components) > 0 {
+				for name := range index.Components {
+					components = append(components, name)
+				}
+			}
+			sort.Strings(components)
+
+			for _, name := range components {
+				options.Components[name] = new([]string)
+				cmd.Flags().StringSliceVar(options.Components[name], name, nil, "Specify the versions for component "+name)
+			}
+			return nil
+		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) < 1 {
-				return cmd.Help()
+			cmd.DisableFlagParsing = false
+			err := cmd.ParseFlags(args)
```

Comment:
It's better to use `SetHelpFunc` to overwrite the help function. The current implementation will cause inconsistent between `tiup help mirror clone` and `tiup mirror clone -h`.

Reply:
addressed, PTAL